src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -285,6 +296,11 @@ internal virtual void Validate()
             {
                 throw new InvalidOperationException(CoreErrorMessages.AdfsNotCurrentlySupportedAuthorityType);
             }
+
+            if (Config.TelemetryCallback != null && Config.MatsConfig != null)
+            {
+                throw new InvalidOperationException(CoreErrorMessages.MatsAndTelemetryCallbackCannotBeConfiguredSimultaneously);
```

Comment:
Should be MsalClientException

Reply:
Should _all_ of the InvalidOperationExceptions be MsalClientException then?  

---
In reply to: [270951122](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/955#discussion_r270951122) [](ancestors = 270951122)

Reply:
We only throw Invalid arg exceptions (mostly ArgNull..) that are not instances of MsalExceptions. Customer apps will catch MsalExceptions and provide some user feedback (e.g. "Auth services down, try again later"). If we start throwing anything else, we will break this promise. 

It's also a matter of consistency, we validate other params as well and throw MsalClientEx